test/test_cuda.py

```diff
@@ -1759,9 +1764,18 @@ def leak_gpu0():
             l.append(torch.randn(1024 * 1024 * 8, device=torch.device("cuda:0")))
 
         no_leak()
-
-        with self.assertRaisesRegex(RuntimeError, r"CUDA driver API confirmed .+ on device 0.+"):
-            leak_gpu0()
+        regex = r"CUDA driver API confirmed .+ on device 0.+"
+        if IS_JETSON:
+            try:
+                leak_gpu0()
+            except RuntimeError as e:
+                import re
```

Comment:
What's the point of this?

Reply:
as explained in the comments below:
```
# assertRaisesRegex does not pass with Python for Jetson,
# even though the RuntimeError matches regex using re.match
```
without my changes, the test fails on jetson

Reply:
That sounds very worrying no? Is the error any different than the original one? Or the string not formatted exactly the same way?

Reply:
https://docs.google.com/document/d/1XaHtyH3sU1ItS-wUkyq89ynqpWUbbasDLn2fiRw_ebE/edit?usp=sharing
this doc steps through my investigation of the issue. the string appears to be formatted identically

Reply:
following up:
```
Alban D
The error here is not that the message doesn't match but that the error is not raised at all.
Rishi Puri US:
i dont think so. the error message is just misleading. see below. If i remove the assert, the test throws the expected runtime error. now if i catch that error and use it for re.match the test passes
```

Reply:
@albanD how should i proceed with this?

Reply:
I'm not sure, there isn't an obvious answer here indeed.
But I feel like there is a pretty simple problem statement and there is no reason for this not to work. So I would expect that we figure out what the problem is here instead of just pushing that under the rug.

But if you want to do that for the jetson-only code, I won't block the PR on this.